metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala

```diff
@@ -121,6 +127,182 @@ final class BloopServers(
     }
   }
 
+  private def writeJVMPropertiesToBloopGlobalJsonFile(
+      bloopGlobalJsonFile: File,
+      requestedBloopJvmProperties: List[String],
+      javaHome: String
+  ): Try[Unit] = Try {
+    val bw = new BufferedWriter(
+      new FileWriter(bloopGlobalJsonFile)
+    )
+    val text = new Gson().toJson(
+      List("javaOptions" -> requestedBloopJvmProperties, "javaHome" -> javaHome)
+    )
+    bw.write(text)
+    bw.close()
+  }
+
+  private def getBloopGlobalJsonLastModifiedByMetalsTime: Long = Try {
+    val source = scala.io.Source.fromFile(
+      Paths
+        .get(System.getProperty("user.home"))
+        .resolve(".bloop/created_by_metals.lock")
+        .toFile
+    )
+    val value: Long = source.mkString.toLong
+    source.close()
+    value
+  }.getOrElse(0)
```

Comment:
```suggestion
  private def getBloopGlobalJsonLastModifiedByMetalsTime: Option[Long] = {
    val lockFilePath = Paths
        .get(System.getProperty("user.home"))
        .resolve(".bloop/created_by_metals.lock")
    val absolutePath = AbsolutePath(lockFilePath)
    if (absolutePath.exists())
       Try(text.toLong).toOption
    else
      None
  }
```

We can then check if it's defined to see if it was MEtals that modified the Bloop file.

Reply:
oops, just realized why the bug existed for not working in the extension...
The for comprehension conditions the execution. of logic to when the bloopGlobalJson file preexisted. So I need to revise that code.